matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/tasks/UploadSignaturesTask.kt

```diff
@@ -36,11 +35,14 @@ internal class DefaultUploadSignaturesTask @Inject constructor(
 
     override suspend fun execute(params: UploadSignaturesTask.Params) {
         try {
-            val response = executeRequest<SignatureUploadResponse>(globalErrorReceiver) {
-                this.isRetryable = true
-                this.maxRetryCount = 10
-                this.apiCall = cryptoApi.uploadSignatures(params.signatures)
-            }
+            val response = executeRequest(
+                    globalErrorReceiver,
+                    { cryptoApi.uploadSignatures(params.signatures) },
+                    {
+                        isRetryable = true
```

Comment:
This would definitely look nicer as a parameter instead of DSL.

Reply:
Very true